common/lib/transport/connectionmanager.js

```diff
@@ -387,14 +387,14 @@ var ConnectionManager = (function() {
 			* be a sync with the new connectionSerial (which will be -1). (And it
 			* needs to be set in the library, which is done by activateTransport). */
 			var connectionReset = connectionId !== self.connectionId,
-				newConnectionSerial = connectionReset ? connectionSerial : self.connectionSerial;
+				syncConnectionSerial = connectionReset ? preUpgradeConnectionSerial : self.connectionSerial;
```

Comment:
`scheduleTransportActivation()` is passed the `connectionSerial` from the `CONNECTED` message when the upgrade transport connects. So how is this the `preUpgradeConnectionSerial` ?

Reply:
In the sense that the sync as the atomic moment of the actual 'upgrade', so preUpgradeConnectionSerial is the serial of the upgrade transport before that (likely -1). Can change to preSyncConnectionSerial if you think that's clearer